cirq/ops/eigen_gate.py

```diff
@@ -186,8 +219,11 @@ def _unitary_(self) -> Union[np.ndarray, type(NotImplemented)]:
         if self.is_parameterized():
             return NotImplemented
         e = cast(float, self._exponent)
-        return np.sum([1j**(half_turns * e * 2) * component for half_turns,
-                       component in self._eigen_components()], axis=0)
+        return np.sum([
+            component * 1j**(
+                    2 * e * (half_turns + self._global_shift_in_half_turns))
+            for half_turns, component in self._eigen_components()
+        ], axis=0)
```

Comment:
What do you think of
```
            component * (-1)**(
                    e * (half_turns + self._global_shift_in_half_turns))
```
instead?

Reply:
Acknowledged, but it doesn't give an exact result in the same cases.

`(-1+0j)**(1*0.5) == (6.123233995736766e-17+1j)`

`1j**(2*0.5) == 1j`

This has an observable impact, namely the X gate's matrix being perturbed away from [[0, 1], [1, 0]] which I find distasteful.